syntax = "proto3";

import "google/protobuf/timestamp.proto";

message Name {
  string first_name = 1;
  string middle_name = 2;
  string last_name = 3;
}

message BackgroundCheckRequest {
  Name name = 1;
  google.protobuf.Timestamp birthdate = 2;

  enum CheckOptions {
    ONEID = 0;
    RCMP = 1;
    // ...
  }

  repeated CheckOptions checks = 3;
}

message BackgroundCheckResponse {
  string id = 1;
}

service BackgroundCheckService {
  rpc CreateBackgroundCheck (BackgroundCheckRequest) returns (BackgroundCheckResponse) {}
}